scripts/cross_version_oncotree_translator.py

```diff
@@ -448,7 +448,7 @@ def write_summary_file(output_file, source_version, target_version):
     resolved_codes = sorted([resolved_code for resolved_code, resolved_node in GLOBAL_LOG_MAP.items() if len(resolved_node[CHOICES_FIELD]) == 1], key = lambda k: sort_by_resolution_method(k, GLOBAL_LOG_MAP[k]))
 
 
-    with open(re.sub("\.[^.]+$", "_summary.html", output_file), "w") as f:
```

Comment:
@averyniceday this seems prone to bugs. Can you set some variable to the html output filename instead?

Reply:
do you mean have a set summary html file name? like always summary_report.html?

Reply:
@ao508 ^

Reply:
@averyniceday create a variable like:

```
output_html_file = re.sub("\.[^.]+$", "_summary.html", output_file)
with open(output_html_file, 'w')
   ....
```

Reply:
or wait do you mean do the filename construction separately outside of the open()

Reply:
@averyniceday if all that's being done is getting the file basename (w/o extension) then you can just 

```
output_html_file = os.path.basename(output_file).split('.')[0] + '_summary.html'
```

Reply:
instead of relying on regex to do the substitution for you

Reply:
@averyniceday 

you would need to add the output directory also... so perhaps:


```
output_html_file  = os.path.basename(output_file).split('.')[0] + '_summary.html'
output_file_path = os.path.join(os.path.dirname(output_file), output_html_file)
```

Reply:
super side edge case but any chance someone passes in an output file name with multiple periods like `case.1.txt`? 
do you prefer using some split/string processing without the regex? I can use something similar

Reply:
@averyniceday use this instead:

```
os.path.splitext(path)
```
>Split the pathname path into a pair (root, ext) such that root + ext == path, and ext is empty or begins with a period and contains at most one period. Leading periods on the basename are ignored; splitext('.cshrc') returns ('.cshrc', '').

see [here](https://docs.python.org/3/library/os.path.html) (scroll to bottom)

Reply:
magic!!! okay adding it in

Reply:
@averyniceday  use this:


```
html_summary_file = os.path.splitext(output_file) + '_summary.html'
```